src/operator/tensor/indexing_op.h

```diff
@@ -493,6 +493,133 @@ void BatchTakeOpForward(const nnvm::NodeAttrs& attrs,
   });
 }
 
+/*!
+ * \brief The parameters of the one_hot operator.
+ */
+struct OneHotParam : public dmlc::Parameter<OneHotParam> {
+  int depth;
+  double on_value;
+  double off_value;
+  int axis;
+  int dtype;
+  DMLC_DECLARE_PARAMETER(OneHotParam) {
+    DMLC_DECLARE_FIELD(depth)
+      .describe("The dimension size at dim = axis.");
+    DMLC_DECLARE_FIELD(on_value)
+      .set_default(1.0f)
+      .describe("The value assigned to the locations represented by indices.");
+    DMLC_DECLARE_FIELD(off_value)
+      .set_default(0.0f)
+      .describe("The value assigned to the locations not represented by indices.");
+    DMLC_DECLARE_FIELD(axis)
+      .set_default(-1)
+      .describe("The dimension position at which a new dimension will be created."
+                " So far, only axis = -1 is supported, which means a new dimension"
+                " is always appended at the end of the array. Setting it to other"
+                " numbers will cause the program to throw an exception");
```

Comment:
I think we can remove the axis here and constrain the one_hot to always insert a new dimension.